src/Models/LoginSession.php

```diff
@@ -186,6 +187,41 @@ private function handlePermission(string $funcName, $member): bool
         return false;
     }
 
+    /**
+     * @param HTTPRequest|null $request
+     * @return string
+     */
+    public static function getAnonymizedIPFromRequest(HTTPRequest $request = null): string
```

Comment:
This whole method feels a big magic - did you use some resource to write this code? Is there an existing library we can use instead of implementing this directly in core?

Reply:
yes, i think we should use HTTPUtils from symfony, but currently, that class has been (partially) copied over to the project. I'm happy to update the HTTPUtils and use that instead
or when/if a move is made to use the symfony/request package, that would be even more easier

Reply:
I can't find a `HTTPUtils` class in framework... where has it been copied over? Do you mean the `HTTP` class?

If Symfony's `HTTPUtils` offers this functionality, and nothing in framework offers equivalent functionality, I'd be in favour of simply adding `HTTPUtils` as a dependency for this PR.

If it is the case that we have a `HTTPUtils` class, or its equivalent, and the API and behaviour is the same for the methods that are shared, I would be in favour of deprecating _our_ class and directing people to use the symfony class instead - though that's going beyond the scope of this PR so that could happily be a separate issue to be dealt with independently.

Reply:
@GuySartorelli ah yes sorry, it's IPUtils not HTTPUtils :-) from the symfony/http-foundation package

Reply:
see https://github.com/silverstripe/silverstripe-framework/pull/10844